src/model/UpsetColumn.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Created by bikramkawan on 24/11/2016.
+ */
+import * as d3 from 'd3';
+import Column, {IColumnDesc, IStatistics} from './Column';
+import ValueColumn from './ValueColumn';
+
+
+export function numberCompare(a: number, b: number) {
+  if (isNaN(a)) { //NaN are bigger
+    return isNaN(b) ? 0 : +1;
+  }
+  if (isNaN(b)) {
+    return -1;
+  }
+  return a - b;
+}
+
+// Calculate Median, Q1 and Q1)
+export function getPercentile(data, percentile) {
+
+  var index = (percentile / 100) * data.length;
+  var result;
+  if (Math.floor(index) === index) {
+    result = (data[(index - 1)] + data[index]) / 2;
+  } else {
+    result = data[Math.floor(index)];
+  }
+  return result;
+}
+
+export class CustomSortCalculation {
```

Comment:
since there is only one valid option. I would inline the code in the compare function. reduces complexity